### DISCUSSION TITLE
Need help about environment variables 😣 (SvelteKit & postgresql)

### DISCUSSION THREAD

**Discussion by essovius** [coollabsio/coolify#4499] [Q&A]
Hello, this is the first time i'm using coolify. I've succesfully installed it on my server. I've created a new project and connected to github. This project is a SvelteKit app. I also created a postgresql database. To deploy app I've followed instructions here: https://svelte.dev/docs/kit/adapter-node and here: https://coolify.io/docs/applications/svelte-kit 

But unfortunately deployments failed after this section: 
```
#13 0.691 > vite build
#13 0.691
#13 2.169 vite v6.0.2 building SSR bundle for production...
#13 2.206 transforming...
#13 17.74 "serial" is imported from external module "drizzle-orm/pg-core" but never used in "src/lib/server/db/schema.ts".
#13 17.74 ✓ 3980 modules transformed.
#13 17.94 rendering chunks...
#13 18.86 
#13 18.86 node:internal/event_target:1062
#13 18.86   process.nextTick(() => { throw err; });
#13 18.86                            ^
#13 18.87 TypeError [ERR_INVALID_URL]: Invalid URL
#13 18.87     at new NodeError (node:internal/errors:405:5)
#13 18.87     at new URL (node:internal/url:676:13)
```

```
#13 18.87   input: '${environment.DATABASE_URL}',
#13 18.87   code: 'ERR_INVALID_URL'
#13 18.87 }
```

I also tried editing environments directly in application but it still gave the same error. 
I only enter two variables: 
* `DATABASE_URL=` starts with `postgres://`
* `ORIGIN=`  generated url in app page that ends with `sslip.io` 

Can someone please help me with this? I couldn't figure out where am i doing wrong. I'm using Nixpacks for building
Thanks in advance

**Comment by stickerdaniel** [comment]
1) Create the DB as a separate Resource in Coolify
2) Add a schema definition under Initialization scripts in the DB config (combine the generated .sql files in your drizzle/ folder)
3) Copy the Postgres URL (internal) into your DATABASE_URL environment variable

https://coolify.io/docs/databases/
